src/intermediate.rs

```diff
@@ -845,133 +954,125 @@ impl RustType {
                 // but other aliases are generated and should be used.
                 AliasIdent::Rust(_) => format!("{}{}", core, ident),
             },
-            RustType::CBORBytes(ty) => ty.for_rust_member(from_wasm),
         }
     }
 
     /// IDENTIFIER for an enum variant. (Use for_rust_member() for the )
     pub fn for_variant(&self) -> VariantIdent {
         match self {
-            RustType::Fixed(f) => f.for_variant(),
-            RustType::Primitive(p) => p.to_variant(),
-            RustType::Rust(ident) => VariantIdent::new_rust(ident.clone()),
-            RustType::Array(inner) => VariantIdent::new_custom(format!("Arr{}", inner.for_variant())),
-            RustType::Tagged(_tag, ty) => ty.for_variant(),
+            Self::Fixed(f) => f.for_variant(),
+            Self::Primitive(p) => p.to_variant(),
+            Self::Rust(ident) => VariantIdent::new_rust(ident.clone()),
+            Self::Array(inner) => VariantIdent::new_custom(format!("Arr{}", inner.conceptual_type.for_variant())),
             // TODO: should we not end up in this situation and just insert a Null fixed value instead?
-            RustType::Optional(ty) => VariantIdent::new_custom(format!("Opt{}", ty.for_variant())),
-            RustType::Map(k, v) => VariantIdent::new_custom(Self::name_for_wasm_map(k, v).to_string()),
-            RustType::Alias(ident, _ty) => match ident {
+            Self::Optional(ty) => VariantIdent::new_custom(format!("Opt{}", ty.conceptual_type.for_variant())),
+            Self::Map(k, v) => VariantIdent::new_custom(Self::name_for_wasm_map(k, v).to_string()),
+            Self::Alias(ident, _ty) => match ident {
                 AliasIdent::Rust(rust_ident) => VariantIdent::new_rust(rust_ident.clone()),
                 AliasIdent::Reserved(reserved) => VariantIdent::new_custom(reserved),
             },
-            RustType::CBORBytes(ty) => VariantIdent::new_custom(format!("CBORBytes{}", ty.for_variant())),
         }
     }
 
     /// for parameter TYPES from wasm that take ownership (via cloning here)
     /// can_fail is for cases where checks (e.g. range checks) are done if there
     /// is a type transformation (i.e. wrapper types) like text (wasm) -> #6.14(text) (rust)
     pub fn from_wasm_boundary_clone(&self, expr: &str, can_fail: bool) -> Vec<ToWasmBoundaryOperations> {
-        //assert!(matches!(self, RustType::Tagged(_, _)) || !can_fail);
+        //assert!(matches!(self, Self::Tagged(_, _)) || !can_fail);
         match self {
-            RustType::Tagged(_tag, ty) => {
-                let mut inner = ty.from_wasm_boundary_clone(expr, can_fail);
-                if can_fail {
-                    inner.push(ToWasmBoundaryOperations::TryInto);
-                } else {
-                    inner.push(ToWasmBoundaryOperations::Into);
-                }
-                inner
-            },
-            RustType::Rust(_ident) => vec![
+            // Self::Tagged(_tag, ty) => {
```

Comment:
what will we do with this code?

Reply:
It will need changing later on. It'll depend how we want to handle wrapper types. Here it'll be for ones that have specific range/etc checks where a conversion between the underlying type could fail. This shouldn't have anything to do with tags but it's an artifact of how we only had wrapper types for tagged stuff before, then later tacked on range checks way later.

Reply:
This logic is now separated from the tagged type. There are however a lot of TODO's in other places from when this was first written as to which places can have can-fail APIs (e.g. range/etc checks). The current state is good enough for our purposes (CML) but at some point we could resolve those. Or change how we handle types that are a subset of their underlying types.